addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyMissingScanRule.java

```diff
@@ -146,4 +96,61 @@ private String getAlertAttribute(String key) {
     public Map<String, String> getAlertTags() {
         return ALERT_TAGS;
     }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return Arrays.asList(
+                alertMissingCspHeader().build(),
+                alertObsoleteCspHeader().build(),
+                alertObsoleteCspHeader().build());
+    }
+
+    private static boolean hasCspHeader(HttpMessage msg) {
+        return !msg.getResponseHeader()
+                .getHeaderValues(HttpFieldsNames.CONTENT_SECURITY_POLICY)
+                .isEmpty();
+    }
+
+    private static boolean hasObsoleteCspHeader(HttpMessage msg) {
+        return !msg.getResponseHeader().getHeaderValues("X-Content-Security-Policy").isEmpty()
+                || !msg.getResponseHeader().getHeaderValues("X-WebKit-CSP").isEmpty();
+    }
+
+    private static boolean hasCspReportOnlyHeader(HttpMessage msg) {
+        return !msg.getResponseHeader()
+                .getHeaderValues("Content-Security-Policy-Report-Only")
+                .isEmpty();
+    }
+
+    private AlertBuilder buildAlert(int risk) {
+        return newAlert()
+                .setRisk(risk)
+                .setConfidence(Alert.CONFIDENCE_HIGH)
+                .setCweId(693) // CWE-693: Protection Mechanism Failure
+                .setWascId(15); // WASC-15: Application Misconfiguration
+    }
+
+    private AlertBuilder alertMissingCspHeader() {
+        return buildAlert(Alert.RISK_MEDIUM)
+                .setName(getAlertAttribute("name"))
+                .setDescription(getAlertAttribute("desc"))
+                .setSolution(getAlertAttribute("soln"))
+                .setReference(getAlertAttribute("refs"));
```

Comment:
Could/should also `.setAlertRef(PLUGIN_ID + "-1");`

Reply:
Worth checking them in the unit tests too :wink:

Thanks for sticking with this. The finish line is in sight 🎉

Reply:
IMO the number could be a parameter of the `buildAlert`, less duplication.